sql/src/main/java/io/crate/execution/dsl/projection/SourceIndexWriterProjection.java

```diff
@@ -35,26 +35,29 @@
 import java.io.IOException;
 import java.util.Arrays;
 import java.util.List;
+import java.util.Objects;
 
 /**
  * IndexWriterProjector that gets its values from a source input
  */
 public class SourceIndexWriterProjection extends AbstractIndexWriterProjection {
 
-    private final Boolean overwriteDuplicates;
+    private static final String OVERWRITE_DUPLICATES = "overwrite_duplicates";
+    private static final boolean OVERWRITE_DUPLICATES_DEFAULT = false;
 
+    private final Boolean overwriteDuplicates;
     private final Reference rawSourceReference;
     private final InputColumn rawSourceSymbol;
-
-    private static final String OVERWRITE_DUPLICATES = "overwrite_duplicates";
-    private static final boolean OVERWRITE_DUPLICATES_DEFAULT = false;
+    private final List<? extends Symbol> outputs;
 
     @Nullable
     private String[] includes;
 
     @Nullable
     private String[] excludes;
 
+    @Nullable
+    private final ReturnSummarySymbols returnSummarySymbols;
```

Comment:
Just a thought: Maybe we should also move into the direction of having different concrete projection classes instead of having a lot of either/or nullable properties. It's hard to comprehend what properties are set in which case and the "either or" nullability cannot be expressed nicely with the type system / annotations